src/windows/gfx/d3d11-ctx.c

```diff
@@ -45,8 +58,185 @@ static void d3d11_ctx_get_size(struct d3d11_ctx *ctx, uint32_t *width, uint32_t
 	*height = rect.bottom - rect.top;
 }
 
+static void mty_validate_format_colorspace(struct d3d11_ctx *ctx, MTY_ColorFormat format, MTY_ColorSpace colorspace, DXGI_FORMAT *format_out, DXGI_COLOR_SPACE_TYPE *colorspace_out)
+{
+	DXGI_FORMAT format_new = DXGI_FORMAT_R8G8B8A8_UNORM;
+	DXGI_COLOR_SPACE_TYPE colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+
+	// Use the last known value if unspecified
+	if (format == MTY_COLOR_FORMAT_UNKNOWN) {
+		format_new = ctx->format;
+	}
+	if (colorspace == MTY_COLOR_SPACE_UNKNOWN) {
+		colorspace_new = ctx->colorspace;
+	}
+
+	switch (format) {
+		case MTY_COLOR_FORMAT_RGBA16F:
+			format_new = DXGI_FORMAT_R16G16B16A16_FLOAT;
+			break;
+		case MTY_COLOR_FORMAT_RGB10A2:
+			format_new = DXGI_FORMAT_R10G10B10A2_UNORM;
+			break;
+	}
+
+	switch (colorspace) {
+		case MTY_COLOR_SPACE_SRGB:
+			colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+			break;
+		case MTY_COLOR_SPACE_SCRGB_LINEAR:
+			colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709;
+			break;
+		case MTY_COLOR_SPACE_HDR10:
+			colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+			break;
+	}
+
+	// Ensure that the format and colorspace are a valid pairing
+	switch (colorspace_new) {
+		case DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709: {
+			if (format_new != DXGI_FORMAT_R16G16B16A16_FLOAT) {
+				MTY_Log("Format 0x%X is not meant for colorspace 0x%X. Forcing format to 0x%X.", format_new, DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709, DXGI_FORMAT_R16G16B16A16_FLOAT);
+				format_new = DXGI_FORMAT_R16G16B16A16_FLOAT;
+			}
+			break;
+		}
+		case DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020: {
+			if (format_new != DXGI_FORMAT_R10G10B10A2_UNORM) {
+				MTY_Log("Format 0x%X is not meant for colorspace 0x%X. Forcing format to 0x%X.", format_new, DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020, DXGI_FORMAT_R10G10B10A2_UNORM);
+				format_new = DXGI_FORMAT_R10G10B10A2_UNORM;
+			}
+			break;
+		}
+		default:
+			break;
+	}
+
+	*format_out = format_new;
+	*colorspace_out = colorspace_new;
+}
+
+static bool d3d11_ctx_refresh_window_bounds(struct d3d11_ctx *ctx)
+{
+	bool changed = false;
+
+	RECT window_bounds_new = {0};
+	GetWindowRect(ctx->hwnd, &window_bounds_new);
+
+	LONG dt_left = window_bounds_new.left - ctx->window_bounds.left;
+	LONG dt_top = window_bounds_new.top - ctx->window_bounds.top;
+	LONG dt_right = window_bounds_new.right - ctx->window_bounds.right;
+	LONG dt_bottom = window_bounds_new.bottom - ctx->window_bounds.bottom;
+
+	changed = labs(dt_left) || labs(dt_top) || labs(dt_right) || labs(dt_bottom);
```

Comment:
Why are we making these absolute? Any value is basically "changed"

```suggestion
	changed = window_bounds_new.left - ctx->window_bounds.left != 0 || window_bounds_new.top - ctx->window_bounds.top != 0
		|| window_bounds_new.right - ctx->window_bounds.right != 0 || window_bounds_new.bottom - ctx->window_bounds.bottom != 0;
```
or
```suggestion
	changed = window_bounds_new.left - ctx->window_bounds.left || window_bounds_new.top - ctx->window_bounds.top
		|| window_bounds_new.right - ctx->window_bounds.right || window_bounds_new.bottom - ctx->window_bounds.bottom;
```
Would also work